### ISSUE TITLE
UGens: Add a DMovingSum UGen (demand-rate moving sum)

### ISSUE THREAD

**Issue by jleben** [supercollider/supercollider#292]
[Issue migrated from SourceForge | ID: 3164823 | Submitted by 'jamshark70']
[http://sourceforge.net/support/tracker.php?aid=3164823]

Using a moving average to smooth jittery data is an important technique. Currently it's rather painful at demand rate -- such as this minimal test case:
```
SynthDef(\movingsum, { |t_trig = 0, n = 6|
	var in, buf, phase, prev, initTrig, fb, sum, writer;

	in = Latch.kr(t_trig, t_trig);
	buf = LocalBuf(n).clear;
	phase = Demand.kr(t_trig, 0, Dseq([Dseries(0, 1, n)], inf));
	prev = Demand.kr(t_trig, 0, Dbufrd(buf, phase));
	initTrig = Impulse.kr(0);
	fb = Select.kr(initTrig, [LocalIn.kr(1), 0]);
	sum = fb + ((t_trig > 0) \* (in - prev));
	writer = Demand.kr(t_trig <! prev, 0, Dbufwr(in, buf, phase));
	
	LocalOut.kr(sum);
	Poll.kr(t_trig, sum);
}).add;
```
It's quite delicate, and error prone to write from scratch every time.

It would be much nicer to be able to write something like this:
```
SynthDef(\movingsum, { |t_trig = 0, n = 6|
	var sum = Demand.kr(t_trig, 0, DMovingSum(t_trig, n));
	Poll.kr(t_trig, sum);
}).add;
```
... which, of course, could be turned into a moving average by `DMovingSum(t_trig, n) / n`.

**Comment by LFSaw** [comment]
so this'd basically be an integrator over the last `n` triggered events?

**Comment by telephon** [comment]
yes, I think so. A leaky integrator in demand rate might also be useful.

**Comment by LFSaw** [comment]
this means to keep a FIFO of past values to pop-subtract before push-adding the new value. ideally, it should be possible to modulate `n`.

**Comment by telephon** [comment]
yes, similar to `DelayN`, with `maxSize` and `size`.

**Comment by mtmccrea** [comment]
Somewhat randomly came across this old issue,

> ideally, it should be possible to modulate n

FWIW `n` is modulatable in `MovingSum` (in sc3-plugins), though it is also not demand-rate.
